sklearn/naive_bayes.py

```diff
@@ -471,10 +471,14 @@ def _update_class_log_prior(self, class_prior=None):
             self.class_log_prior_ = np.zeros(n_classes) - np.log(n_classes)
 
     def _check_alpha(self):
-        if self.alpha < 0:
+        if np.min(self.alpha) < 0:
             raise ValueError('Smoothing parameter alpha = %.1e. '
-                             'alpha should be > 0.' % self.alpha)
-        if self.alpha < _ALPHA_MIN:
+                             'alpha should be > 0.' % np.min(self.alpha))
+        if isinstance(self.alpha, np.ndarray):
+            if not self.alpha.shape[0] == self.feature_count_.shape[1]:
+                raise ValueError("alpha should be a scalar or a numpy array "
+                                 "with shape [n_features]")
+        if np.min(self.alpha) < _ALPHA_MIN:
             warnings.warn('alpha too small will result in numeric errors, '
                           'setting alpha = %.1e' % _ALPHA_MIN)
             return _ALPHA_MIN
```

Comment:
If `alpha = np.array([1e-11, 1])`, alpha will be set to `1e-10`. It should be `np.array([1e-10, 1])` instead, isn't it?

```python
if np.min(self.alpha) < _ALPHA_MIN:
    warnings.warn('alpha too small will result in numeric errors, '
                  'setting alpha = %.1e' % _ALPHA_MIN)
    alpha_ = alpha.copy()
    alpha_[alpha_ < _ALPHA_MIN] = _ALPHA_MIN
    return alpha_
```

Reply:
Yes, seems better than just changing to a non-vector pseudocount of _ALPHA_MIN. I have made changes accordingly and added a test.